com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/metadata/util/MtaMetadataUtil.java

```diff
@@ -45,11 +47,16 @@ public static boolean hasMtaMetadata(CloudEntity entity) {
                                              .keySet();
         Set<String> metadataAnnotations = metadata.getAnnotations()
                                                   .keySet();
-        return metadataLabels.containsAll(MTA_METADATA_LABELS) && metadataAnnotations.containsAll(MTA_METADATA_ANNOTATIONS);
+        return metadataLabels.containsAll(MTA_METADATA_MANDATORY_LABELS)
+            && metadataAnnotations.containsAll(MTA_METADATA_MANDATORY_ANNOTATIONS);
     }
 
-    public static String getHashedMtaId(String mtaId) {
-        return DigestUtils.md5DigestAsHex(mtaId.getBytes());
+    public static String getHashedLabel(String mtaLabel) {
+        if (StringUtils.isEmpty(mtaLabel)) {
```

Comment:
null values cannot be hashed, so that's why we catch them and an empty string doesn't really need hashing in the first place (we're hashing for length concerns). The null value is handled down the line as it could have a meaning of 'label doesn't exist'